fix `enum-compare-switch` warning

**Steps to reproduce**
Build with clang-13 @ macosx
```
clang -v
Apple clang version 13.0.0 (clang-1300.0.29.30)
Target: x86_64-apple-darwin20.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

**What is the expected behavior?**
no compile warning

**What is the actual behavior?**
a lot of `enum-compare-switch` warning
```
$ ninja
[1/2] Building CXX object exporters/otlp/CMakeFiles/opentelemetry_exporter_otlp_grpc_client.dir/src/otlp_grpc_utils.cc.o
/Users/flier/github/opentelemetry-cpp/exporters/otlp/src/otlp_grpc_utils.cc:53:10: warning: comparison of different enumeration types in switch statement ('::grpc::StatusCode' and 'grpc_status_code') [-Wenum-compare-switch]
    case GRPC_STATUS_DATA_LOSS:
         ^~~~~~~~~~~~~~~~~~~~~
/Users/flier/github/opentelemetry-cpp/exporters/otlp/src/otlp_grpc_utils.cc:51:10: warning: comparison of different enumeration types in switch statement ('::grpc::StatusCode' and 'grpc_status_code') [-Wenum-compare-switch]
    case GRPC_STATUS_UNAVAILABLE:
         ^~~~~~~~~~~~~~~~~~~~~~~
/Users/flier/github/opentelemetry-cpp/exporters/otlp/src/otlp_grpc_utils.cc:49:10: warning: comparison of different enumeration types in switch statement ('::grpc::StatusCode' and 'grpc_status_code') [-Wenum-compare-switch]
    case GRPC_STATUS_INTERNAL:
         ^~~~~~~~~~~~~~~~~~~~
/Users/flier/github/opentelemetry-cpp/exporters/otlp/src/otlp_grpc_utils.cc:47:10: warning: comparison of different enumeration types in switch statement ('::grpc::StatusCode' and 'grpc_status_code') [-Wenum-compare-switch]
    case GRPC_STATUS_UNIMPLEMENTED:
         ^~~~~~~~~~~~~~~~~~~~~~~~~
/Users/flier/github/opentelemetry-cpp/exporters/otlp/src/otlp_grpc_utils.cc:45:10: warning: comparison of different enumeration types in switch statement ('::grpc::StatusCode' and 'grpc_status_code') [-Wenum-compare-switch]
    case GRPC_STATUS_OUT_OF_RANGE:
         ^~~~~~~~~~~~~~~~~~~~~~~~
/Users/flier/github/opentelemetry-cpp/exporters/otlp/src/otlp_grpc_utils.cc:43:10: warning: comparison of different enumeration types in switch statement ('::grpc::StatusCode' and 'grpc_status_code') [-Wenum-compare-switch]
    case GRPC_STATUS_ABORTED:
         ^~~~~~~~~~~~~~~~~~~
/Users/flier/github/opentelemetry-cpp/exporters/otlp/src/otlp_grpc_utils.cc:41:10: warning: comparison of different enumeration types in switch statement ('::grpc::StatusCode' and 'grpc_status_code') [-Wenum-compare-switch]
    case GRPC_STATUS_FAILED_PRECONDITION:
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/flier/github/opentelemetry-cpp/exporters/otlp/src/otlp_grpc_utils.cc:39:10: warning: comparison of different enumeration types in switch statement ('::grpc::StatusCode' and 'grpc_status_code') [-Wenum-compare-switch]
    case GRPC_STATUS_RESOURCE_EXHAUSTED:
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/flier/github/opentelemetry-cpp/exporters/otlp/src/otlp_grpc_utils.cc:37:10: warning: comparison of different enumeration types in switch statement ('::grpc::StatusCode' and 'grpc_status_code') [-Wenum-compare-switch]
    case GRPC_STATUS_UNAUTHENTICATED:
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/flier/github/opentelemetry-cpp/exporters/otlp/src/otlp_grpc_utils.cc:35:10: warning: comparison of different enumeration types in switch statement ('::grpc::StatusCode' and 'grpc_status_code') [-Wenum-compare-switch]
    case GRPC_STATUS_PERMISSION_DENIED:
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/flier/github/opentelemetry-cpp/exporters/otlp/src/otlp_grpc_utils.cc:33:10: warning: comparison of different enumeration types in switch statement ('::grpc::StatusCode' and 'grpc_status_code') [-Wenum-compare-switch]
    case GRPC_STATUS_ALREADY_EXISTS:
         ^~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/flier/github/opentelemetry-cpp/exporters/otlp/src/otlp_grpc_utils.cc:31:10: warning: comparison of different enumeration types in switch statement ('::grpc::StatusCode' and 'grpc_status_code') [-Wenum-compare-switch]
    case GRPC_STATUS_NOT_FOUND:
         ^~~~~~~~~~~~~~~~~~~~~
/Users/flier/github/opentelemetry-cpp/exporters/otlp/src/otlp_grpc_utils.cc:29:10: warning: comparison of different enumeration types in switch statement ('::grpc::StatusCode' and 'grpc_status_code') [-Wenum-compare-switch]
    case GRPC_STATUS_DEADLINE_EXCEEDED:
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/flier/github/opentelemetry-cpp/exporters/otlp/src/otlp_grpc_utils.cc:27:10: warning: comparison of different enumeration types in switch statement ('::grpc::StatusCode' and 'grpc_status_code') [-Wenum-compare-switch]
    case GRPC_STATUS_INVALID_ARGUMENT:
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/flier/github/opentelemetry-cpp/exporters/otlp/src/otlp_grpc_utils.cc:25:10: warning: comparison of different enumeration types in switch statement ('::grpc::StatusCode' and 'grpc_status_code') [-Wenum-compare-switch]
    case GRPC_STATUS_UNKNOWN:
         ^~~~~~~~~~~~~~~~~~~
/Users/flier/github/opentelemetry-cpp/exporters/otlp/src/otlp_grpc_utils.cc:23:10: warning: comparison of different enumeration types in switch statement ('::grpc::StatusCode' and 'grpc_status_code') [-Wenum-compare-switch]
    case GRPC_STATUS_CANCELLED:
         ^~~~~~~~~~~~~~~~~~~~~
/Users/flier/github/opentelemetry-cpp/exporters/otlp/src/otlp_grpc_utils.cc:21:10: warning: comparison of different enumeration types in switch statement ('::grpc::StatusCode' and 'grpc_status_code') [-Wenum-compare-switch]
    case GRPC_STATUS_OK:
         ^~~~~~~~~~~~~~
17 warnings generated.
[2/2] Linking CXX static library exporters/otlp/libopentelemetry_exporter_otlp_grpc_client.a
```

--- a/exporters/otlp/src/otlp_grpc_utils.cc
+++ b/exporters/otlp/src/otlp_grpc_utils.cc
@@ -16,39 +16,39 @@ const char *grpc_status_code_to_string(::grpc::StatusCode status_code)
 {
   switch (status_code)
   {
-    case GRPC_STATUS_OK:
+    case ::grpc::StatusCode::OK:
       return "OK";
-    case GRPC_STATUS_CANCELLED:
+    case ::grpc::StatusCode::CANCELLED:
       return "CANCELLED";
-    case GRPC_STATUS_UNKNOWN:
+    case ::grpc::StatusCode::UNKNOWN:
       return "UNKNOWN";
-    case GRPC_STATUS_INVALID_ARGUMENT:
+    case ::grpc::StatusCode::INVALID_ARGUMENT:
       return "INVALID_ARGUMENT";
-    case GRPC_STATUS_DEADLINE_EXCEEDED:
+    case ::grpc::StatusCode::DEADLINE_EXCEEDED:
       return "DEADLINE_EXCEEDED";
-    case GRPC_STATUS_NOT_FOUND:
+    case ::grpc::StatusCode::NOT_FOUND:
       return "NOT_FOUND";
-    case GRPC_STATUS_ALREADY_EXISTS:
+    case ::grpc::StatusCode::ALREADY_EXISTS:
       return "ALREADY_EXISTS";
-    case GRPC_STATUS_PERMISSION_DENIED:
+    case ::grpc::StatusCode::PERMISSION_DENIED:
       return "PERMISSION_DENIED";
-    case GRPC_STATUS_UNAUTHENTICATED:
+    case ::grpc::StatusCode::UNAUTHENTICATED:
       return "UNAUTHENTICATED";
-    case GRPC_STATUS_RESOURCE_EXHAUSTED:
+    case ::grpc::StatusCode::RESOURCE_EXHAUSTED:
       return "RESOURCE_EXHAUSTED";
-    case GRPC_STATUS_FAILED_PRECONDITION:
+    case ::grpc::StatusCode::FAILED_PRECONDITION:
       return "FAILED_PRECONDITION";
-    case GRPC_STATUS_ABORTED:
+    case ::grpc::StatusCode::ABORTED:
       return "ABORTED";
-    case GRPC_STATUS_OUT_OF_RANGE:
+    case ::grpc::StatusCode::OUT_OF_RANGE:
       return "OUT_OF_RANGE";
-    case GRPC_STATUS_UNIMPLEMENTED:
+    case ::grpc::StatusCode::UNIMPLEMENTED:
       return "UNIMPLEMENTED";
-    case GRPC_STATUS_INTERNAL:
+    case ::grpc::StatusCode::INTERNAL:
       return "INTERNAL";
-    case GRPC_STATUS_UNAVAILABLE:
+    case ::grpc::StatusCode::UNAVAILABLE:
       return "UNAVAILABLE";
-    case GRPC_STATUS_DATA_LOSS:
+    case ::grpc::StatusCode::DATA_LOSS:
       return "DATA_LOSS";
     default:
       return "UNKNOWN";

